### PR TITLE
Feature/data submission workflow

### PR DESCRIPTION
**Pull request by fiedosiukr**
This PR creates a new section that allows authors to upload additional approvals. In order to submit a project with access policy set to credentialed, either ethical approval or an explanation must be included.

image

### CODE REVIEW

File: physionet-django/project/forms.py [hunk 1]
```diff
@@ -1041,3 +1047,52 @@ def save(self):
...
+class EthicsForm(forms.ModelForm):
+    class Meta:
+        model = ActiveProject
+        fields = ('ethics_statement',)
+        help_texts = {
+            'ethics_statement': (
+                '* A statement regarding ethical concerns for the work. '
+                'This statement will be published with the resource, '
+                'and typically describes formal approvals acquired for '
+                'the creation of the resource, and/or ethical considerations '
```
Comment (alistairewj) [inline physionet-django/project/forms.py]: change to:

"the creation of the resource (such as a review by an ethics board) ..."

File: physionet-django/project/modelcomponents/activeproject.py [hunk 2]
```diff
@@ -563,7 +575,7 @@ def publish(self, slug=None, make_zip=True, title=None):
 
                 return published_project
 
-        except:
+        except Exception:
```
Comment (alistairewj) [inline physionet-django/project/modelcomponents/activeproject.py]: you may run into trouble for this one based off previous conversations!

File: physionet-django/project/modelcomponents/submission.py [hunk 3]
```diff
@@ -32,7 +62,8 @@ class EditLog(models.Model):
         'pn_suitable': 'Is the content suitable for PhysioNet?',
         'editor_comments': 'Comments to authors',
         'no_phi': 'Is the project free of protected health information?',
-        'data_machine_readable': 'Are all files machine-readable?'
+        'data_machine_readable': 'Are all files machine-readable?',
+        'ethics_included': 'Does the submission contain the necessary approvals?',
```
Comment (alistairewj) [inline physionet-django/project/modelcomponents/submission.py]: change to:

'Does the submission contain an appropriate statement regarding ethical concerns, and if necessary, supporting documentation?'

File: physionet-django/project/modelcomponents/activeproject.py [hunk 4]
```diff
@@ -310,6 +313,11 @@ def check_integrity(self):
                 l = self.LABELS[self.resource_type.id][attr] if attr in self.LABELS[self.resource_type.id] else attr.title().replace('_', ' ')
                 self.integrity_errors.append('Missing required field: {0}'.format(l))
 
+        # Approvals
+        if self.access_policy == 2:
```
Comment (alistairewj) [inline physionet-django/project/modelcomponents/activeproject.py]: you for sure remember this from the meeting but just noting it anyway to remove the access policy restriction

File: physionet-django/project/forms.py [hunk 5]
```diff
@@ -1041,3 +1047,52 @@ def save(self):
...
+class UploadedDocumentFormSet(BaseGenericInlineFormSet):
+    form_name = 'project-uploadeddocument-content_type-object_id'
+    item_label = 'Supporting Documents'
+    max_forms = 3
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.help_text = (
+            'Upload supporting documents. For more information on '
+            'what documents are expected, see the [author guidance page].'
```
Comment (alistairewj) [inline physionet-django/project/forms.py]: can the [author guidance page] be changed to a hyperlink to the author guidance page? currently it's here: https://physionet.org/about/publish/#guidelines

also, add this sentence at the end: "These documents reviewed by the editorial staff, and are not published with the project."

**Review by alistairewj — CHANGES_REQUESTED** [review]
Here are the suggestions I promised in the call. Also, can you add the following to the publish section of the author guidelines, below the "Version" bullet point in the "Creating the project metadata" header:

* **Ethics**: Provide a statement regarding the ethical concerns of the resource. This statement should include any formal ethical approval acquired in the curation of the resource, as well as any concerns individuals should be aware of when using the resource. The ethics statement will be published with the project description. Include an approval letter from the respective ethics board and the associated study protocol if ethics approval was acquired (these documents will be reviewed by the editor, but they will not be published with the resource).

File: physionet-django/project/modelcomponents/activeproject.py (hunk 2 above)
Comment (fiedosiukr) [reply physionet-django/project/modelcomponents/activeproject.py]: Yeah, I know, I forgot to push the final changes that I made. I will do it tomorrow.

File: physionet-django/project/forms.py (hunk 1 above)
Comment (fiedosiukr) [reply physionet-django/project/forms.py]: 🆗

File: physionet-django/project/modelcomponents/submission.py (hunk 3 above)
Comment (fiedosiukr) [reply physionet-django/project/modelcomponents/submission.py]: 🆗

File: physionet-django/project/modelcomponents/activeproject.py (hunk 4 above)
Comment (fiedosiukr) [reply physionet-django/project/modelcomponents/activeproject.py]: 🆗

File: physionet-django/project/forms.py (hunk 5 above)
Comment (fiedosiukr) [reply physionet-django/project/forms.py]: 🆗

**Pull request merged**